search-app/src/cmr/search/site/data.clj

```diff
@@ -0,0 +1,40 @@
+(ns cmr.search.site.data
+  "The functions of this namespace are specifically responsible for generating
+  data structures to be consumed by site page templates.
+
+  Of special note: this namespace and its sibling `page` namespace are only
+  ever meant to be used in the `cmr.search.site` namespace, particularly in
+  support of creating site routes for access in a browser.
+
+  Under no circumstances should `cmr.search.site.data` be accessed from outside
+  this context; the data functions defined herein are specifically for use
+  in page templates, structured explicitly for their needs.")
+
+(defn get-index
+  "Return the data for the index page (none for now)."
+  [request]
+  {})
+
+(defn get-landing-links
```

Comment:
That's just organizational -- in the event that we add more than just EOSDIS-tagged collections, there's a home for them to link from, without having to change any URLs/lose any ground in search results ...

As for the rest of your comment, yes -- those are supported. In summary, there are three types of page presented:

1) a top-level home for all landing pages - provides a link to eosdis-tagged collection landing pages
2) a sub-level home for  just eosdis-tagged landing pages - provides links to all providers that meet the criteria
3) a provider-level page that provides the actual links

Since this is really not for users, but just for crawling, I figured we didn't need to worry about minimizing clicks, and the extra organization might work in our favour later ...